### PR TITLE
Add bodies count after discovery scan.

### PR DESCRIPTION
**Pull request by spadino**

### CODE REVIEW

**Comment by robbyxp1** [comment]
sn could be null, you need to protect ;-)

**Comment by klightspeed** [comment]
Why reference the system node when you have the FSSDiscoveryScan from which TotalBodies is set?

**Comment by spadino** [comment]
@robbyxp1 , I included a null check.

@klightspeed, what you suggest? How would you do that?

**Comment by klightspeed** [comment]
>> Why reference the system node when you have the FSSDiscoveryScan from which TotalBodies is set?
>
> what you suggest? How would you do that?

```
var je = he.journalEntry as JournalFSSDiscoveryScan;
if (je != null)
{
    var bodies_found = je.BodyCount;
    SetControlText(bodies_found + " bodies found.");
}
```

File: EDDiscovery/UserControls/Overlays/UserControlSurveyor.cs [hunk 1]
```diff
@@ -126,9 +120,19 @@ private void NewEntry(HistoryEntry he, HistoryList hl)
...
             {
+                SetControlText(string.Empty);
                 pictureBoxSurveyor.ClearImageList();
                 Refresh();
-            }            
+            }
+
+            if (he.EntryType == JournalTypeEnum.FSSDiscoveryScan)
+            {
+                if (sn.TotalBodies != null)
```
Comment (klightspeed) [inline EDDiscovery/UserControls/Overlays/UserControlSurveyor.cs]:
```
sn?.TotalBodies
```

**Pull request merged**